src/Tree/Entity/Repository/NestedTreeRepository.php

```diff
@@ -835,25 +908,124 @@ public function verify()
     }
 
     /**
-     * NOTE: flush your entity manager after
+     * Tries to recover the tree, avoiding entity object hydration and using DQL
+     *
+     * NOTE: DQL UPDATE statements are ported directly into a Database UPDATE statement and therefore bypass any locking
+     * scheme, events and do not increment the version column. Entities that are already loaded into the persistence
+     * context will NOT be synced with the updated database state.
+     * It is recommended to call EntityManager#clear() and retrieve new instances of any affected entity.
+     *
+     * @phpstan-param array{sortByField?: string, sortDirection?: string} $options
+     *
+     * $options['sortByField']   = (string) Sort siblings by specified field while recovering (default: null)
+     * $options['sortDirection'] = (ASC|DESC) The order to sort siblings in, when sortByField is specified (default: 'ASC')
+     *
+     * @throws ORMException
+     */
+    public function recoverFast(array $options = []): void
+    {
+        $defaultOptions = [
+            'sortByField' => null,
+            'sortDirection' => 'ASC',
+        ];
+        $options += $defaultOptions;
+
+        $meta = $this->getClassMetadata();
+        $config = $this->listener->getConfiguration($this->getEntityManager(), $meta->name);
+        $em = $this->getEntityManager();
+
+        $updateQb = $em->createQueryBuilder()
+                       ->update($meta->getName(), 'node')
+                       ->set('node.'.$config['left'], ':left')
+                       ->set('node.'.$config['right'], ':right')
+                       ->where('node.id = :id');
+        if (isset($config['level'])) {
+            $updateQb->set('node.'.$config['level'], ':level');
+        }
+
+        $doRecover = function ($root, &$count, $level) use ($meta, $em, $options, $updateQb, &$doRecover) {
```

Comment:
```suggestion
        $doRecover = function ($root, &$count, $level) use ($meta, $em, $options, $updateQb, &$doRecover): void {
```
We could also add parameter type declarations